hardware/i2c/master/i2c_mcp23017_ecmd.c

```diff
@@ -0,0 +1,428 @@
+/*
+ *
+ * Copyright (c) 2015 Michael Brakemeier <michael@brakemeier.de>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ *
+ * For more information on the GPL, please go to:
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include <stdint.h>
+#include <string.h>
+
+#include <avr/io.h>
+#include <avr/pgmspace.h>
+
+#include "config.h"
+
+#include "core/debug.h"
+#include "protocols/ecmd/ecmd-base.h"
+
+#include "i2c_mcp23017.h"
+
+#ifdef DEBUG_I2C
+#define DEBUG_MCP23017(fnc, msg...) debug_printf("I2C: %s: ", fnc); debug_printf(msg)
+#else
+#define DEBUG_MCP23017(fnc, msg...)
+#endif
+
+/*
+ * Prototypes
+ */
+int16_t parse_cmd_i2c_mcp23017_setreg(char *cmd, char *output, uint16_t len);
```

Comment:
Nope, wrong. Never do this for ecmd parse_cmd_xy()! ecmd_magic.m4 generates prototypes for meta.c from the ecmd_feature() declaration. Actually these prototypes are not required and exist only for cleanliness.